agent/src/protocols/mod.rs

```diff
@@ -80,13 +85,39 @@ fn inner_get_discovery_handler(
     query: &impl EnvVarQuery,
 ) -> Result<Box<dyn DiscoveryHandler + Sync + Send>, Error> {
     match discovery_handler_config {
+        #[cfg(feature = "onvif")]
         ProtocolHandler::onvif(onvif) => Ok(Box::new(onvif::OnvifDiscoveryHandler::new(&onvif))),
+        #[cfg(feature = "udevf")]
         ProtocolHandler::udev(udev) => Ok(Box::new(udev::UdevDiscoveryHandler::new(&udev))),
+        #[cfg(feature = "opcua")]
         ProtocolHandler::opcua(opcua) => Ok(Box::new(opcua::OpcuaDiscoveryHandler::new(&opcua))),
+        #[cfg(feature = "coapcore")]
+        ProtocolHandler::coapcore(coapcore) => Ok(Box::new(
+            coap_core::CoAPCoREDiscoveryHandler::new(&coapcore),
+        )),
         ProtocolHandler::debugEcho(dbg) => match query.get_env_var("ENABLE_DEBUG_ECHO") {
             Ok(_) => Ok(Box::new(debug_echo::DebugEchoDiscoveryHandler::new(dbg))),
             _ => Err(failure::format_err!("No protocol configured")),
         },
+        config => {
+            log::info!("No handler found for configuration {:?}, the default NoopDiscoveryHandler will be used", config);
```

Comment:
when you rebase this code, can you replace the `panic!("No handler found for configuration {:?}", config);` that is in here with `Err(failure::format_err!(No handler found for configuration {:?}", config))`